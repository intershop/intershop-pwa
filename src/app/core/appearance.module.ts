import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
