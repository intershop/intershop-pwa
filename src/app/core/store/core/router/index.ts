// API to access ngrx core router state
export * from './router.selectors';
export * from './router.operators';
