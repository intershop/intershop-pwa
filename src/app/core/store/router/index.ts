export * from './router.actions';
export * from './router.selectors';
export { CoreState } from '../core.state';
