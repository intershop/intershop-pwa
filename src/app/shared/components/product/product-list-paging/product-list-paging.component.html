<ng-container *ngIf="currentPage && pageIndices.length > 1">
  <ul class="product-list-paging">
    <li>
      <ng-container *ngIf="currentPage > 1; else inactiveFirstLink">
        <a
          class="product-list-paging-previous"
          [routerLink]="[]"
          queryParamsHandling="merge"
          [queryParams]="{ page: currentPage - 1 }"
          [fragment]="fragmentOnRouting"
          >&laquo;</a
        >
      </ng-container>
      <ng-template #inactiveFirstLink> &laquo; </ng-template>
    </li>
    <li *ngFor="let p of pageIndices">
      <a
        [ngClass]="{ active: p.value === currentPage }"
        [routerLink]="[]"
        queryParamsHandling="merge"
        [queryParams]="{ page: p.value }"
        [fragment]="fragmentOnRouting"
        >{{ p.display }}</a
      >
    </li>
    <li class="pagination-list-next">
      <ng-container *ngIf="pageIndices[pageIndices.length - 1].value !== currentPage; else inactiveLastLink">
        <a
          [routerLink]="[]"
          queryParamsHandling="merge"
          [queryParams]="{ page: currentPage + 1 }"
          [fragment]="fragmentOnRouting"
          >&raquo;</a
        >
      </ng-container>
      <ng-template #inactiveLastLink> &raquo; </ng-template>
    </li>
  </ul>
</ng-container>
