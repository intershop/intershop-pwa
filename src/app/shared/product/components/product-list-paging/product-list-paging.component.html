<ng-container *ngIf="currentPage">
  <div class="product-list-paging row justify-content-center">
    <ng-container *ngIf="currentPage > 1; else inactiveFirstLink">
      <a class="col-1" [routerLink]="[]" queryParamsHandling="merge" [queryParams]="{ page: currentPage - 1 }"
        >&laquo;</a
      >
    </ng-container>
    <ng-template #inactiveFirstLink>
      <a class=" col-1 inactive">&laquo;</a>
    </ng-template>

    <div class="col-auto">
      <a
        *ngFor="let p of pageIndices"
        [ngClass]="{ active: p.value === currentPage }"
        [routerLink]="[]"
        queryParamsHandling="merge"
        [queryParams]="{ page: p.value }"
        >{{ p.display }}</a
      >
    </div>

    <ng-container *ngIf="pageIndices[pageIndices.length - 1].value !== currentPage; else inactiveLastLink">
      <a class="col-1" [routerLink]="[]" queryParamsHandling="merge" [queryParams]="{ page: currentPage + 1 }"
        >&raquo;</a
      >
    </ng-container>
    <ng-template #inactiveLastLink>
      <a class="col-1 inactive">&raquo;</a>
    </ng-template>
  </div>
</ng-container>
