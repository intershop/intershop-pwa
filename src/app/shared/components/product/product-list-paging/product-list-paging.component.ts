import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

/**
 * Displays paging information for robots.
 *
 * @example
 *  <ish-product-list-paging *ngIf="displayPaging$ | async"
 *    [currentPage]="currentPage$ | async"
 *    [pageIndices]="pageIndices$ | async"
 *  ></ish-product-list-paging>
 */
@Component({
  selector: 'ish-product-list-paging',
  templateUrl: './product-list-paging.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductListPagingComponent {
  @Input({ required: true }) currentPage: number;
  @Input({ required: true }) pageIndices: { value: number; display: string }[];
  @Input() fragmentOnRouting: string;
}
