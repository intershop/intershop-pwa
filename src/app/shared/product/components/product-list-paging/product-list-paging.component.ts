import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

/**
 * Displays paging information for robots.
 *
 * @example
 *  <ish-product-list-paging *ngIf="displayPaging$ | async"
 *    [currentPage]="currentPage$ | async"
 *    [pageIndices]="pageIndices$ | async"
 *  ></ish-product-list-paging>
 */
@Component({
  selector: 'ish-product-list-paging',
  templateUrl: './product-list-paging.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductListPagingComponent {
  @Input() currentPage: number;
  @Input() pageIndices: { value: number; display: string }[];
  @Input() fragmentOnRouting: string;
}
