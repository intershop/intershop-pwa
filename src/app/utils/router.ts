import { ActivatedRoute, ChildActivationEnd, Router } from '@angular/router';
import { merge, Observable } from 'rxjs';
import { distinctUntilChanged, filter, pluck } from 'rxjs/operators';

/**
 * resolves data from child routes
 */
export function resolveChildRouteData<T>(route: ActivatedRoute, router: Router, key: string): Observable<T> {
  return merge(
    // get first data from injected activated route
    route.firstChild.data.pipe(pluck(key)),
    // listen to router events to get other child updates
    router.events.pipe(
      filter(event => event instanceof ChildActivationEnd),
      pluck('snapshot', 'data', key),
      filter((x: T) => !!x)
    )
  ).pipe(distinctUntilChanged());
}
