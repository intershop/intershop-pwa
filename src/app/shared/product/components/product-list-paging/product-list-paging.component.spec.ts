import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { ProductListPagingComponent } from './product-list-paging.component';

describe('Product List Paging Component', () => {
  let component: ProductListPagingComponent;
  let fixture: ComponentFixture<ProductListPagingComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [ProductListPagingComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ProductListPagingComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    component.pageIndices = [1, 2, 3];
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
  });

  it('should render adequately for starting page', () => {
    component.currentPage = 1;
    fixture.detectChanges();

    expect(element).toMatchInlineSnapshot(`
      <div class="product-list-paging">
        <a class="active" ng-reflect-router-link="" href="/?page=1">1</a
        ><a ng-reflect-router-link="" href="/?page=2">2</a
        ><a ng-reflect-router-link="" href="/?page=3">3</a
        ><a ng-reflect-router-link="" href="/?page=2">»</a>
      </div>
    `);
  });

  it('should render adequately for middle page', () => {
    component.currentPage = 2;
    fixture.detectChanges();

    expect(element).toMatchInlineSnapshot(`
      <div class="product-list-paging">
        <a ng-reflect-router-link="" href="/?page=1">«</a
        ><a ng-reflect-router-link="" href="/?page=1">1</a
        ><a class="active" ng-reflect-router-link="" href="/?page=2">2</a
        ><a ng-reflect-router-link="" href="/?page=3">3</a
        ><a ng-reflect-router-link="" href="/?page=3">»</a>
      </div>
    `);
  });

  it('should render adequately for last page', () => {
    component.currentPage = 3;
    fixture.detectChanges();

    expect(element).toMatchInlineSnapshot(`
      <div class="product-list-paging">
        <a ng-reflect-router-link="" href="/?page=2">«</a
        ><a ng-reflect-router-link="" href="/?page=1">1</a
        ><a ng-reflect-router-link="" href="/?page=2">2</a
        ><a class="active" ng-reflect-router-link="" href="/?page=3">3</a>
      </div>
    `);
  });
});
