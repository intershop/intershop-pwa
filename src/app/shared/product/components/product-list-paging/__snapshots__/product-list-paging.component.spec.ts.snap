// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List Paging Component should render adequately for last page 1`] = `
<div
  id="root3"
>
  
  
  <div
    class="product-list-paging"
  >
    
    <a
      href="/some/followup/link?page=2"
    >
      «
    </a>
    
    
    <a
      href="/some/followup/link?page=1"
      ng-reflect-ng-class="[object Object]"
    >
      1
    </a>
    
    <a
      href="/some/followup/link?page=2"
      ng-reflect-ng-class="[object Object]"
    >
      2
    </a>
    
    <a
      class="active"
      href="/some/followup/link?page=3"
      ng-reflect-ng-class="[object Object]"
    >
      3
    </a>
    
  </div>
</div>
`;

exports[`Product List Paging Component should render adequately for middle page 1`] = `
<div
  id="root2"
>
  
  
  <div
    class="product-list-paging"
  >
    
    <a
      href="/some/followup/link?page=1"
    >
      «
    </a>
    
    
    <a
      href="/some/followup/link?page=1"
      ng-reflect-ng-class="[object Object]"
    >
      1
    </a>
    
    <a
      class="active"
      href="/some/followup/link?page=2"
      ng-reflect-ng-class="[object Object]"
    >
      2
    </a>
    
    <a
      href="/some/followup/link?page=3"
      ng-reflect-ng-class="[object Object]"
    >
      3
    </a>
    
    <a
      href="/some/followup/link?page=3"
    >
      »
    </a>
  </div>
</div>
`;

exports[`Product List Paging Component should render adequately for starting page 1`] = `
<div
  id="root1"
>
  
  
  <div
    class="product-list-paging"
  >
    
    
    
    <a
      class="active"
      href="/some/followup/link?page=1"
      ng-reflect-ng-class="[object Object]"
    >
      1
    </a>
    
    <a
      href="/some/followup/link?page=2"
      ng-reflect-ng-class="[object Object]"
    >
      2
    </a>
    
    <a
      href="/some/followup/link?page=3"
      ng-reflect-ng-class="[object Object]"
    >
      3
    </a>
    
    <a
      href="/some/followup/link?page=2"
    >
      »
    </a>
  </div>
</div>
`;
